From 2f3ae3b30cf8ce8cd04d512ebb4c8b2218563085 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Mon, 17 Jun 2024 18:32:07 -0700
Subject: [PATCH 6/6] Load dependent libs globally

I'm not 100% sure why, but we need all dependent shared libs to be loaded
globally.
---
 src/library_dylink.js | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/library_dylink.js b/src/library_dylink.js
index e0fa4721c..502bc7b25 100644
--- a/src/library_dylink.js
+++ b/src/library_dylink.js
@@ -880,11 +880,15 @@ var LibraryDylink = {
 
     // now load needed libraries and the module itself.
     if (flags.loadAsync) {
+      const origGlobal = flags.global;
+      flags.global = true;
       return metadata.neededDynlibs
         .reduce((chain, dynNeeded) => chain.then(() =>
           loadDynamicLibrary(dynNeeded, flags)
         ), Promise.resolve())
-        .then(loadModule);
+        .then(() => {
+          flags.global = origGlobal;
+        }).then(loadModule);
     }
 
     metadata.neededDynlibs.forEach((needed) => loadDynamicLibrary(needed, flags, localScope));
-- 
2.34.1

